core/blockchain.go

```diff
@@ -666,6 +666,8 @@ func (self *BlockChain) InsertChain(chain types.Blocks) (int, error) {
 			PutTransactions(self.chainDb, block, block.Transactions())
 			// store the receipts
 			PutReceipts(self.chainDb, receipts)
+			// Write map map bloom filters
+			WriteMipmapBloom(self.chainDb, block.NumberU64(), receipts)
```

Comment:
You mean canonical number mapping, right? If yes, the yes, that's also a correct place to pt this, but in its current position here, it's wrong. The least it should also be added here https://github.com/obscuren/go-ethereum/blob/mipmap-bloom/core/blockchain.go#L759, if you want to go down the route of only marking upon canonical status.